master/buildbot/steps/artifact.py

```diff
@@ -402,6 +472,113 @@ def start(self):
         self.setCommand(command)
         ShellCommand.start(self)
 
+    @defer.inlineCallbacks
+    def _getBuildRequest(self):
+        triggeredbybrid = self.build.requests[0].id
+        br = yield self.master.db.buildrequests.getBuildRequestTriggered(triggeredbybrid, self.artifactBuilderName)
+        defer.returnValue(br)
+
+
+class DownloadArtifactsFromChildren(LoggingBuildStep, CompositeStepMixin):
+    name = "Download Artifact(s) from triggered builds"
+    description="Downloading artifact(s) from triggered builds..."
+    descriptionDone="Downloaded artifact(s) from triggered builds"
+
+    def __init__(self,
+                 artifactServer,
+                 artifactServerDir,
+                 artifactBuilderName,
+                 workDir='',
+                 artifactServerPort=None,
+                 artifactDestination=None,
+                 artifactDirectory=None,
+                 usePowerShell=True,
+                 **kwargs):
+        self.workDir = workDir
+        self.artifactBuilderName = artifactBuilderName
+        self.artifactDirectory = artifactDirectory
+        self.artifactServer = artifactServer
+        self.artifactServerDir = artifactServerDir
+        self.artifactServerPort = artifactServerPort
+        self.artifactDestination = artifactDestination
+        self.master = None
+        self.usePowerShell = usePowerShell
+        LoggingBuildStep.__init__(self, **kwargs)
+
+    @defer.inlineCallbacks
+    def start(self):
+        if self.master is None:
+            self.master = self.build.builder.botmaster.parent
+
+        self._setUpLogs()
+
+        partitionRequests = yield self.master.db.buildrequests.getBuildRequestsTriggeredBy(self.build.requests[0].id, self.artifactBuilderName)
```

Comment:
if the partitions are never merge by different builders then is ok, otherwise we are going to hit problems when 2 different builders had a partition merged and try to download.